### DISCUSSION TITLE
Why is the weight calculation dependent on `c` and `r`

### DISCUSSION THREAD

**Discussion by xquery-go** [xquery-go/pallet-staking#7] [Q&A]
in this code. `weights.rs`
```rust
fn new_session(r: u32, c: u32, ) -> Weight {
		// Proof Size summary in bytes:
		//  Measured:  `363 + c * (92 ±0)`
		//  Estimated: `7087 + c * (2507 ±0) + r * (2832 ±0)`
		// Minimum execution time: 32_040_000 picoseconds.
		Weight::from_parts(32_511_000, 7087)
			// Standard Error: 309_600
			.saturating_add(Weight::from_parts(15_326_198, 0).saturating_mul(c.into()))
			.saturating_add(T::DbWeight::get().reads(5_u64))
			.saturating_add(T::DbWeight::get().reads((2_u64).saturating_mul(c.into())))
			.saturating_add(T::DbWeight::get().writes((1_u64).saturating_mul(c.into())))
			.saturating_add(Weight::from_parts(0, 2507).saturating_mul(c.into()))
			.saturating_add(Weight::from_parts(0, 2832).saturating_mul(r.into()))
}
```

**Comment by slycompiler** [comment]
The parameters `c` and `r` appear to represent variables that scale the execution time and resource usage of the session. For example, `c` may represent the number of validators or some measure of computation, while `r` could be related to the number of resources required to establish a session. The weight is calculated in a dynamic way, increasing based on the values of `c` and `r`, to ensure that larger or more complex operations consume proportionally more resources.

**Reply by xquery-go** [reply]
How could `c` and `r` be determined in practice, and what might they represent in the context of a blockchain system?

**Reply by slycompiler** [reply]
In practice, `c` and `r` are likely to be parameters that scale based on the complexity and size of the operation within the blockchain. For example:

- `c` might represent the number of participants or validators in a session, or perhaps the number of accounts or transactions processed. As the session involves more participants or more transactions, the complexity increases, and so does the weight.
- `r` could represent the number of resources required for the session, such as the number of unique cryptographic proofs, data chunks, or memory needed for the operation.
